### PR TITLE
ImmutableDB: use the ChainDB's registry instead of an internal one

### PR DESCRIPTION
**Pull request by mrBliss**
This fixes master.

See the separate commits for more details

### CODE REVIEW

File: ouroboros-consensus/src/Ouroboros/Consensus/Node/Run/Mock.hs [hunk 1]
```diff
@@ -38,7 +42,8 @@ instance ( ProtocolLedgerView (SimpleBlock SimpleMockCrypto ext)
   nodeBlockMatchesHeader  = matchesSimpleHeader
   nodeBlockFetchSize      = fromIntegral . simpleBlockSize . simpleHeaderStd
   nodeIsEBB               = const Nothing
-  nodeEpochSize           = \_ _ _ -> return 100
+  nodeEpochSize           = \_ cfg _ -> return $
+    EpochSize $ 10 * maxRollbacks (protocolSecurityParam cfg)
```
Comment (nfrisby) [inline ouroboros-consensus/src/Ouroboros/Consensus/Node/Run/Mock.hs]: Would it be more consistent to rely on `Cardano.Chain.ProtocolConstants.kEpochSlots` here?

Comment (edsko) [reply ouroboros-consensus/src/Ouroboros/Consensus/Node/Run/Mock.hs]: The mock ledger should I think be able to make independent decisions?

Comment (nfrisby) [reply ouroboros-consensus/src/Ouroboros/Consensus/Node/Run/Mock.hs]: I agree it should, but do we want it to here? My intuition is to minimize differences between tests and production where possible.

Comment (edsko) [reply ouroboros-consensus/src/Ouroboros/Consensus/Node/Run/Mock.hs]: ... Or should we avoid having the code too dependent on specific values..? :)

Comment (nfrisby) [reply ouroboros-consensus/src/Ouroboros/Consensus/Node/Run/Mock.hs]: Well right now they're both `10 * k` :)

Comment (edsko) [reply ouroboros-consensus/src/Ouroboros/Consensus/Node/Run/Mock.hs]: Fair point, let's change that ;D

**Pull request merged**